client/src/index.js

```diff
@@ -45,9 +78,55 @@ class MainView extends React.Component {
   }
 }
 
-ReactDOM.render(
-  <div className={css('feedback-app-main-container')}>
-    <MainView />
-  </div>,
-  document.getElementById('root')
-)
+let isInitialized = false
+
+const initialize = () => {
+  if (isInitialized || !document.body) {
+    return
+  }
+
+  isInitialized = true
+
+  const loadPersist = async (state, allDataLoaded) => {
+    store.dispatch({ type: LOAD_PERSIST, state })
+
+    if (allDataLoaded) {
+      if (!state.users || R.isEmpty(state.users)) {
+        const response = await fetch(`${apiUrl}/users`, {
+          method: 'POST',
+        })
+        const { id, secret } = await response.json()
+
+        console.log('Created new user from API', { [id]: secret })
+
+        store.dispatch({
+          type: SET_PERSIST,
+          data: {
+            users: {
+              [id]: secret,
+            },
+          },
+        })
+      } else {
+        console.log('Loaded user from persistent storage', state.users)
+      }
+    }
+  }
+
+  const savePersist = setupPersist(loadPersist)
+
+  store.subscribe(() => {
+    savePersist(store.getState().persist || { })
+  })
+
+  ReactDOM.render(
+    <div className={css('feedback-app-main-container')}>
+      <MainView />
+    </div>,
+    feedbackAppRoot()
+  )
+}
+
+initialize()
+document.addEventListener('DOMContentLoaded', initialize)
+window.addEventListener('load', initialize)
```

Comment:
This is not reliable, google or read e.g. https://stackoverflow.com/questions/588040/window-onload-vs-document-onload

Does not have to be changed now but maybe rethought at some point?

Reply:
Shouldn't hurt as `initialize()` works only on the first call